tutorregistry/plugin.py

```diff
@@ -0,0 +1,80 @@
+from __future__ import annotations
+
+import click
+import json
+
+from tutor import config as tutor_config
+from tutor import exceptions, hooks, utils
+from tutor.commands.k8s import k8s
+
+from .__about__ import __version__
+
+
+hooks.Filters.CONFIG_DEFAULTS.add_items(
+    [
+        ("REGISTRY_VERSION", __version__),
+    ]
+)
+
+hooks.Filters.CONFIG_UNIQUE.add_items(
+    [
+        ("REGISTRY_CREDENTIALS", {}),
+    ]
+)
+
+
+@k8s.command(help="Configure access to a private registry with an "
+                  "imagePullSecret")
+@click.pass_obj
+def registry(context) -> None:
+    """
+    Create imagePullSecret(s) for private registries and
+    add them to the default ServiceAccount in the given namespace.
+    If a secret with the same name already exists delete and
+    recreate it, so credentials are always up to date for the secrets.
+    run CMD --> tutor k8s registry
+    """
+    config = tutor_config.load(context.root)
+    namespace = config["K8S_NAMESPACE"]
+    creds = config["REGISTRY_CREDENTIALS"]
+
+    all_keys = {"imagePullSecrets": []}
+
+    for registry in list(creds.keys()):
+        key_name = creds[registry].get("key_name", "SET_ME")
+        username = creds[registry].get("username", "SET_ME")
+        password = creds[registry].get("password", "SET_ME")
+        # email is optional; depending on the registry
+        email = creds[registry].get("email", "")
```

Comment:
To the best of my knowledge, you can iterate through the keys in a dictionary like so:

```python
for registry in creds:
```

But in this case, perhaps you instead want to use:

```suggestion
    for registry, cred in creds.items():
        key_name = cred.get("key_name", "SET_ME")
        username = cred.get("username", "SET_ME")
        password = cred.get("password", "SET_ME")
        # email is optional; depending on the registry
        email = cred.get("email", "")
```

Reply:
Fixed in the latest change :)